### PR TITLE
Prevent migrated deals from reaching TFM

### PR DESCRIPTION
**Pull request by IgnatG**
- added a validation that checks whether the deal has the `dataMigration` property
- if a deal has the `dataMigration` property, then it shouldn't create a record in TFM and it shouldn't trigger ACBS, Estore, etc

### CODE REVIEW

File: portal-api/src/v1/gef/controllers/application.controller.js [hunk 1]
```diff
@@ -272,28 +263,17 @@ exports.changeStatus = async (req, res) => {
...
 
   // If status of correct type, send update email
-  if (
-    [
-      DEAL_STATUS.READY_FOR_APPROVAL,
-      DEAL_STATUS.CHANGES_REQUIRED,
-      DEAL_STATUS.SUBMITTED_TO_UKEF,
-    ].includes(status)
-  ) {
+  if ([DEAL_STATUS.READY_FOR_APPROVAL, DEAL_STATUS.CHANGES_REQUIRED, DEAL_STATUS.SUBMITTED_TO_UKEF].includes(status)) {
```
Comment (abhi-markan) [inline portal-api/src/v1/gef/controllers/application.controller.js:271]: Should we leave it in separate lines, as more than 3 should be in separate lines (if in future new statuses are added).

File: trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js [hunk 2]
```diff
@@ -163,20 +165,60 @@ const submitDealAfterUkefIds = async (dealId, dealType, checker) => {
...
+  const { dealId, dealType, checker } = req.body;
   let deal;
 
-  const canSubmitDealAfterUkefIds = await dealHasAllUkefIds(dealId);
-
-  if (canSubmitDealAfterUkefIds) {
-    deal = await submitDealAfterUkefIds(dealId, dealType, checker);
+  const { status: canSubmitDealAfterUkefIds, message } = await dealHasAllUkefIds(dealId);
+  // check if the deal has been migrated from Portal V1 to Portal v2
+  if (message === 'Migrated deal') {
```
Comment (abhi-markan) [inline trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js:212]: Constant?

**Review by abhi-markan — CHANGES_REQUESTED** [review]
Minor comments

File: trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js (hunk 2 above)
Comment (IgnatG) [reply trade-finance-manager-api/src/v1/controllers/deal.submit.controller.js:212]: I don't think we need it for this one. It's only used here and we will remove it once we finish amendments.

**Review by abhi-markan — APPROVED** [review]

**Pull request merged**